src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -122,58 +122,55 @@ function adjustStudentsPanelView(panelHeading, panelCollapse, isDataHandsontable
  */
-function expandCollapseExistingStudentsPanel() {
-    const panelHeading = $(this);
-    const panelCollapse = panelHeading.parent().children('.panel-collapse');
-
-    const ajaxPreloadImage = panelHeading.parent().find('#ajax-preload-image');
-    const ajaxStatusMessage = panelHeading.parent().find('.ajax-status-message');
-    ajaxStatusMessage.hide(); // hide any status message from the previous state
 
-    // perform AJAX only if existing students' spreadsheet is empty
-    if (getSpreadsheetLength(dataHandsontable.getData()) === 0) {
-        getAjaxStudentList(ajaxPreloadImage)
-                .then((data) => {
-                    ajaxPreloadImage.toggle();
-                    if (data.students.length === 0) {
+function studentsPanelClickHandler(currentSpreadsheet) {
+    const panelHeading = $(currentSpreadsheet.currentTarget);
+    const panelCollapse = panelHeading.parent().children('.panel-collapse');
+    const isExistingStudentsPanel = panelCollapse.find('#enrollSpreadsheet').length === 1;
+
+    if (isExistingStudentsPanel) { // Existing students' panel
+        const ajaxPreloadImage = panelHeading.find('#ajax-preload-image');
+        const ajaxStatusMessage = panelHeading.find('.ajax-status-message');
+        ajaxStatusMessage.hide(); // hide any status message from the previous state
+
+        // perform AJAX only if existing students' spreadsheet is empty
+        if (getSpreadsheetLength(dataHandsontable.getData()) === 0) {
+            getAjaxStudentList(ajaxPreloadImage)
+                    .then((data) => {
+                        ajaxPreloadImage.toggle();
+                        if (data.students.length === 0) {
+                            ajaxStatusMessage.toggle();
+                            ajaxStatusMessage.text('No existing students in course.');
+                        } else {
+                            loadExistingStudentsData(data.students);
+                            adjustStudentsPanelView(panelHeading, panelCollapse, true);
+                        }
+                    }).catch(() => {
+                        ajaxPreloadImage.toggle();
                         ajaxStatusMessage.toggle();
-                        ajaxStatusMessage.text('No existing students in course.');
-                    } else {
-                        loadExistingStudentsData(data.students);
-                        adjustStudentsPanelView(panelHeading, panelCollapse, true);
-                    }
-                }).catch(() => {
-                    ajaxPreloadImage.toggle();
-                    ajaxStatusMessage.toggle();
-                    ajaxStatusMessage.text('Failed to load. Click here to retry.');
-                });
-    } else {
+                        ajaxStatusMessage.text('Failed to load. Click here to retry.');
+                    });
+        } else {
+            adjustStudentsPanelView(panelHeading, panelCollapse, true);
```

Comment:
To reduce the level of nested if else and improve readability, a better pattern is to put this short block at the start and use early return statement, i.e.
```javascript
if (spreadsheet_not_empty) {
    adjust...();
    return;
}
// code for loading existing student data
```

Reply:
Turns out this is an ESLint restriction `no-negated-condition`